contactsummary/js/src/contact-summary/controllers/keyDetails.js

```diff
@@ -14,37 +15,52 @@ define([
    * @param {ContractService} Contract
    * @constructor
    */
-  function KeyDetailsCtrl($log, ContactDetails, Contract) {
+  function KeyDetailsCtrl($log, ContactDetails, Contract, pubSubService) {
     $log.debug('Controller: KeyDetailsCtrl');
 
     this.ready = false;
 
-    ContactDetails.get()
-      .then(function (response) {
-        this.contactDetails = response;
-        return Contract.getPrimary();
-      }.bind(this))
-      .then(function (response) {
-        if (_.isEmpty(response)) {
-          return;
+
+
+    var getContacts = function(){
```

Comment:
If we do that it wont have access to `$log, ContactDetails, Contract, pubSubService` variables.